src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -102,11 +102,6 @@
 
         gb.withRateLimitHandler(CUSTOMIZED);
         OkHttpClient client = new OkHttpClient().setProxy(getProxy(defaultIfBlank(apiUrl, GITHUB_URL)));
-        client.setCache(GitHubClientCacheOps.toCacheDir().apply(config));
-        if (config.getClientCacheSize() > 0) {
-            Cache cache = toCacheDir().apply(config);
-            client.setCache(cache);
-        }
```

Comment:
> two consecutive webhooks could trigger two calls to `github.getRepositories()` without any cache invalidation in between (because both cache invalidation calls are previous to the API call)

Not sure I follow.

1. first commit made
1. first webhook received
1. cache invalidated
1. second commit made
1. second webhook received
1. cache invalidated
1. `retrieve` runs, asks GH without cache, sees second commit, builds it
1. `retrieve` runs again, loads cached response of second commit, decides correctly that nothing has changed

> If two branch indexing are manually triggered more or less near in time and a commit happens in the middle, the cache issue will hit again.

This is true. It does not affect people using only scheduled polling (with a schedule interval greater than the cache expiry time!), but could be seen as a bug by someone going to Jenkins and manually forcing reindexing after a particular commit.